shell/platform/android/io/flutter/embedding/engine/mutatorsstack/FlutterMutatorView.java

```diff
@@ -88,8 +103,27 @@ public void dispatchDraw(Canvas canvas) {
     // all the clipping paths
     finalMatrix.postTranslate(-left, -top);
 
-    canvas.concat(finalMatrix);
-    super.dispatchDraw(canvas);
-    canvas.restore();
+    return finalMatrix;
   }
+
+  @Override
+  // Intercept the events here and do not propagate them to the child platform views.
+  public boolean onInterceptTouchEvent(MotionEvent event) {
+    return true;
+  }
+
+  @Override
+  public boolean onTouchEvent(MotionEvent event) {
+    if (androidTouchProcessor == null) {
+      return super.onTouchEvent(event);
+    }
+
+    // Mutator view itself doesn't rotate, scale, skew, etc.
+    // we only need to account for translation.
+    Matrix screenMatrix = new Matrix();
+    screenMatrix.postTranslate(getLeft(), getTop());
```

Comment:
did you intend to use `getPlatformViewMatrix()` here or somewhere else?

Reply:
nope, it is for when we need to support mutations on top of platform views

Reply:
the mutator view does rotate, translate and scale. right @cyanglaz ?